Heisenbug in test suite

**Describe the bug**

There's a couple tests that sometimes fail because of *reasons*. Probably there's some test-isolation issue, or something related to the factories that leads to these problems. We should figure out what's happening and why.

**To Reproduce**

No deterministic way of reproducing, yet.

**Expected behavior**

Tests should always pass or always fail.

Example builds: 
- https://github.com/open-zaak/open-zaak/runs/1563843491?check_suite_focus=true
- https://github.com/open-zaak/open-zaak/runs/1562686399?check_suite_focus=true

Failing tests:

```
======================================================================
Adding content type 'auth | group'
FAIL: test_partial_update_enkelvoudiginformatieobject_audittrail (openzaak.components.documenten.tests.test_audittrails.AuditTrailTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/open-zaak/open-zaak/src/openzaak/components/documenten/tests/test_audittrails.py", line 195, in test_partial_update_enkelvoudiginformatieobject_audittrail
    informatieobject_partial_update_audittrail.actie, "partial_update"
AssertionError: 'create' != 'partial_update'
- create
+ partial_update


======================================================================
FAIL: test_update_enkelvoudiginformatieobject_audittrail (openzaak.components.documenten.tests.test_audittrails_cmis.AuditTrailTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/open-zaak/open-zaak/src/openzaak/components/documenten/tests/test_audittrails_cmis.py", line 162, in test_update_enkelvoudiginformatieobject_audittrail
    self.assertEqual(informatieobject_update_audittrail.actie, "update")
AssertionError: 'create' != 'update'
- create
+ update
```

--- a/src/openzaak/components/documenten/tests/test_audittrails.py
+++ b/src/openzaak/components/documenten/tests/test_audittrails.py
@@ -184,13 +184,21 @@ class AuditTrailTests(JWTAuthMixin, APITestCase):
             {"titel": "changed", "lock": "0f60f6d2d2714c809ed762372f5a363a"},
         ).data
 
-        audittrails = AuditTrail.objects.filter(hoofd_object=informatieobject_url)
+        audittrails = AuditTrail.objects.filter(
+            hoofd_object=informatieobject_url
+        ).order_by("pk")
         self.assertEqual(audittrails.count(), 2)
 
         # Verify that the audittrail for the EnkelvoudigInformatieObject
         # partial update contains the correct information
         informatieobject_partial_update_audittrail = audittrails[1]
         self.assertEqual(informatieobject_partial_update_audittrail.bron, "DRC")
+
+        # XXX: tracking down the Heisenbug
+        if informatieobject_partial_update_audittrail.actie != "partial_update":
+            print(audittrails[0].__dict__)
+            print(audittrails[1].__dict__)
+
         self.assertEqual(
             informatieobject_partial_update_audittrail.actie, "partial_update"
         )

--- a/src/openzaak/components/documenten/tests/test_audittrails_cmis.py
+++ b/src/openzaak/components/documenten/tests/test_audittrails_cmis.py
@@ -152,7 +152,9 @@ class AuditTrailTests(JWTAuthMixin, APICMISTestCase):
 
         informatieobject_response = self.client.put(informatieobject_url, content).data
 
-        audittrails = AuditTrail.objects.filter(hoofd_object=informatieobject_url)
+        audittrails = AuditTrail.objects.filter(
+            hoofd_object=informatieobject_url
+        ).order_by("pk")
         self.assertEqual(audittrails.count(), 2)
 
         # Verify that the audittrail for the EnkelvoudigInformatieObject update

